bin/cask

```diff
@@ -324,10 +330,21 @@ def main():
             exit_error(
                 'Python 2.6 required, yours is {0.major}.{0.minor}'.format(
                     sys.version_info))
+
+
         if len(sys.argv) > 1 and sys.argv[1] == 'exec':
+            ## never returns!
             exec_command(sys.argv[2:])
-        else:
-            exec_cask(sys.argv[1:])
+
+        ## special handling for --emacs/bin as this has to happen before
+        ## everything else, and we don't want the parsing in two places.
+        if len(sys.argv) > 1:
+            if sys.argv[1] == '--emacs':
+                global EMACS
```

Comment:
@phillord Well, it definitely makes sense to call it only once, but the result should not be stored in global state.  As I said, once you have determined the proper Emacs executable, either pass it through via function arguments, or introduce a new class to wrap this state.

> Yes, I agree that the CASK variables are not altered once set. The EMACS variable would work in the same way. 

I'd not…

> The only question is where the initialization happens. The obvious place being main(), yes.

… because the point about the current globals is not that they are set just once.  The point is—and that is why they are *constants* really—that they are set once and only once *at definition*, and at no other place.  And that's the only kind of global a Python program should have in my opinion.